### PR TITLE
Add IR Metrics

### LINKED ISSUE #43: More Ranking Metrics
There are several metrics listed here that evaluate the quality of a ranking:

https://en.wikipedia.org/wiki/Learning_to_rank

Let's add them to our metrics.

### PR DESCRIPTION
**Pull request by orionw**
- Adds AP, mAP, DCG, and NDCG
- Tests for those metrics 

Closes #43

### CODE REVIEW

**Comment by orionw** [comment]
Why default to exponential for the NDCG metric?

**Comment by bjschoenfeld** [comment]
Your docstrings said exponential was the default, so I started thinking about it. Then I noticed on the wikipedia article that it says:

- "[It] places stronger emphasis on retrieving relevant documents."
- "The [exponential] formula is commonly used in industry including major web search companies and data science competition platforms such as Kaggle."

I guess it is not really going to be that big of a difference because our numbers are between 0 and 1.

**Comment by orionw** [comment]
I would prefer to keep MAP in the code, even if we don't use it.  It is still a valid metric.

**Comment by bjschoenfeld** [comment]
I made [another branch](https://github.com/byu-dml/d3m-dynamic-neural-architecture/tree/mean-avg-precision) so we don't lose it.

File: dna/problems.py [hunk 1]
```diff
@@ -339,6 +347,7 @@ def score(self, predictions_by_group, targets):
         top_1_regrets = []
         top_k_regrets = []
         top_k_counts = []
+        # TODO: ndcg@k
```
Comment (bjschoenfeld) [inline dna/problems.py]: Can we add ndcg to the subset problem? I don't have time right now, but I may later this evening.

Comment (orionw) [reply dna/problems.py]: Sure, I don't see why not.  We don't have to use it if we don't want to.

**Pull request merged**